### PR TITLE
Allows passing an OpenTelemetry instance to registerObservers() methods.

### PR DESCRIPTION
**Pull request by jenarros**
Resolves open-telemetry/opentelemetry-java-instrumentation#5715

### CODE REVIEW

File: instrumentation/oshi/library/src/main/java/io/opentelemetry/instrumentation/oshi/ProcessMetrics.java [hunk 1]
```diff
@@ -21,8 +22,10 @@ private ProcessMetrics() {}
 
   /** Register observers for java runtime metrics. */
   public static void registerObservers() {
-    // TODO(anuraaga): registerObservers should accept an OpenTelemetry instance
```
Comment (anuraaga) [inline instrumentation/oshi/library/src/main/java/io/opentelemetry/instrumentation/oshi/ProcessMetrics.java:24]: Thanks for helping on this! Let's remove these methods and have the callers pass in a opentelemetry. The javaagent will pass in GlobalOpenTelemetry.get and library instrumentation tests will pass in `testing.getOpenTelemetry()` if I'm not mistaken

Comment (jenarros) [reply instrumentation/oshi/library/src/main/java/io/opentelemetry/instrumentation/oshi/ProcessMetrics.java:24]: Should I add `@Deprecated` to the methods first to give time for users to update their apps and then in a different release we remove them?

Comment (trask) [reply instrumentation/oshi/library/src/main/java/io/opentelemetry/instrumentation/oshi/ProcessMetrics.java:24]: > Should I add `@Deprecated` to the methods first to give time for users to update their apps and then in a different release we remove them?

that would be great 👍

Comment (trask) [reply instrumentation/oshi/library/src/main/java/io/opentelemetry/instrumentation/oshi/ProcessMetrics.java:24]: oh, I see you did already, thx!

**Review by trask — APPROVED** [review]

**Review by anuraaga — APPROVED** [review]
Thanks!

**Pull request merged**